caffe2/CMakeLists.txt

```diff
@@ -1310,6 +1330,14 @@ if(BUILD_MOBILE_TEST)
   endforeach()
 endif()
 
+add_executable(lite_interpreter_demo ${TORCH_ROOT}/test/cpp/demos/main.cpp)
+target_link_libraries(lite_interpreter_demo torch_library)
+if(ADD_DEBUG_INFO)
+  print_target_properties(torch)
+  print_target_properties(lite_interpreter_demo)
+  print_target_properties(torch_cpu)
+endif()
```

Comment:
Can you please elaborate, why this is needed?

Reply:
I used it when debugging libtorch and libtorch_cpu targets. The previous debug if statement only prints the file list, but it's not clear whether the target includes the file list correctly. This function can examine the target so it's more accurate. I attached an example below and hopefully it's clearer why it helps. The `-- CPU sources: ` comes from the previous debug if statement, and the `torch_cpu ` comes from the debug helper function `print_target_properties` in this pr.

```
-- Using third party subdirectory Eigen.
-- Using third_party/pybind11.
-- pybind11 include dirs: /Users/chenlai/pytorch/cmake/../third_party/pybind11/include
-- Version: 7.0.3
-- Build type: Release
-- CXX_STANDARD: 14
-- Required features: cxx_variadic_templates
-- Not using libkineto in a mobile build.
-- Could NOT find Backtrace (missing: Backtrace_LIBRARY Backtrace_INCLUDE_DIR)
-- don't use NUMA
-- Using ATen parallel backend: NATIVE
disabling CUDA because USE_CUDA is set false
AT_INSTALL_INCLUDE_DIR include/ATen/core
core header install: /Users/chenlai/pytorch/build_android/aten/src/ATen/core/TensorBody.h
-- CPU sources: 
--   /Users/chenlai/pytorch/aten/src/ATen/BatchedFallback.cpp
--   /Users/chenlai/pytorch/aten/src/ATen/BatchedTensorImpl.cpp
--   /Users/chenlai/pytorch/aten/src/ATen/BatchingRegistrations.cpp
--   /Users/chenlai/pytorch/aten/src/ATen/CPUGeneratorImpl.cpp
--   /Users/chenlai/pytorch/aten/src/ATen/Context.cpp
--   /Users/chenlai/pytorch/aten/src/ATen/DLConvertor.cpp
--   /Users/chenlai/pytorch/aten/src/ATen/DynamicLibrary.cpp
--   /Users/chenlai/pytorch/aten/src/ATen/ExpandUtils.cpp
--   /Users/chenlai/pytorch/aten/src/ATen/LegacyTHFunctionsCPU.cpp
--   /Users/chenlai/pytorch/aten/src/ATen/MemoryOverlap.cpp
--   /Users/chenlai/pytorch/aten/src/ATen/NamedTensorUtils.cpp
--   /Users/chenlai/pytorch/aten/src/ATen/ParallelCommon.cpp
--   /Users/chenlai/pytorch/aten/src/ATen/ParallelNative.cpp
--   /Users/chenlai/pytorch/aten/src/ATen/ParallelNativeTBB.cpp=
...
torch_cpu CXX_STANDARD = 14
torch_cpu C_STANDARD = 11
torch_cpu IMPORTED = FALSE
torch_cpu IMPORTED_GLOBAL = FALSE
torch_cpu INCLUDE_DIRECTORIES = /Users/chenlai/pytorch/build_android/aten/src;/Users/chenlai/pytorch/aten/src;/Users/chenlai/pytorch/build_android;/Users/chenlai/pytorch;/Users/chenlai/pytorch/third_party/XNNPACK/include;/Users/chenlai/pytorch/cmake/../third_party/eigen;/Users/chenlai/pytorch/cmake/../third_party/pybind11/include;/Users/chenlai/pytorch/caffe2/aten/src/TH;/Users/chenlai/pytorch/build_android/caffe2/aten/src/TH;/Users/chenlai/pytorch/aten/src;/Users/chenlai/pytorch/build_android/caffe2/aten/src;/Users/chenlai/pytorch/build_android/aten/src;/Users/chenlai/pytorch/aten/src;/Users/chenlai/pytorch/build_android/caffe2/../aten/src;/Users/chenlai/pytorch/build_android/caffe2/../aten/src/ATen;/Users/chenlai/pytorch/build_android/aten/src;/Users/chenlai/pytorch/torch/csrc;/Users/chenlai/pytorch/third_party/miniz-2.0.8;/Users/chenlai/pytorch/build_android/caffe2/aten/src/TH;/Users/chenlai/pytorch/aten/src/TH;/Users/chenlai/pytorch/aten/src;/Users/chenlai/pytorch/build_android/caffe2/aten/src;/Users/chenlai/pytorch/build_android/aten/src;/Users/chenlai/pytorch/aten/src;/Users/chenlai/pytorch/aten/../third_party/catch/single_include;/Users/chenlai/pytorch/aten/src/ATen/..;/Users/chenlai/pytorch/build_android/caffe2/aten/src/ATen;/Users/chenlai/pytorch/third_party/miniz-2.0.8;/Users/chenlai/pytorch/caffe2/
torch_cpu INCLUDE_DIRECTORIES = /Users/chenlai/pytorch/build_android/aten/src;/Users/chenlai/pytorch/aten/src;/Users/chenlai/pytorch/build_android;/Users/chenlai/pytorch;/Users/chenlai/pytorch/third_party/XNNPACK/include;/Users/chenlai/pytorch/cmake/../third_party/eigen;/Users/chenlai/pytorch/cmake/../third_party/pybind11/include;/Users/chenlai/pytorch/caffe2/aten/src/TH;/Users/chenlai/pytorch/build_android/caffe2/aten/src/TH;/Users/chenlai/pytorch/aten/src;/Users/chenlai/pytorch/build_android/caffe2/aten/src;/Users/chenlai/pytorch/build_android/aten/src;/Users/chenlai/pytorch/aten/src;/Users/chenlai/pytorch/build_android/caffe2/../aten/src;/Users/chenlai/pytorch/build_android/caffe2/../aten/src/ATen;/Users/chenlai/pytorch/build_android/aten/src;/Users/chenlai/pytorch/torch/csrc;/Users/chenlai/pytorch/third_party/miniz-2.0.8;/Users/chenlai/pytorch/build_android/caffe2/aten/src/TH;/Users/chenlai/pytorch/aten/src/TH;/Users/chenlai/pytorch/aten/src;/Users/chenlai/pytorch/build_android/caffe2/aten/src;/Users/chenlai/pytorch/build_android/aten/src;/Users/chenlai/pytorch/aten/src;/Users/chenlai/pytorch/aten/../third_party/catch/single_include;/Users/chenlai/pytorch/aten/src/ATen/..;/Users/chenlai/pytorch/build_android/caffe2/aten/src/ATen;/Users/chenlai/pytorch/third_party/miniz-2.0.8;/Users/chenlai/pytorch/caffe2/
torch_cpu INCLUDE_DIRECTORIES = /Users/chenlai/pytorch/build_android/aten/src;/Users/chenlai/pytorch/aten/src;/Users/chenlai/pytorch/build_android;/Users/chenlai/pytorch;/Users/chenlai/pytorch/third_party/XNNPACK/include;/Users/chenlai/pytorch/cmake/../third_party/eigen;/Users/chenlai/pytorch/cmake/../third_party/pybind11/include;/Users/chenlai/pytorch/caffe2/aten/src/TH;/Users/chenlai/pytorch/build_android/caffe2/aten/src/TH;/Users/chenlai/pytorch/aten/src;/Users/chenlai/pytorch/build_android/caffe2/aten/src;/Users/chenlai/pytorch/build_android/aten/src;/Users/chenlai/pytorch/aten/src;/Users/chenlai/pytorch/build_android/caffe2/../aten/src;/Users/chenlai/pytorch/build_android/caffe2/../aten/src/ATen;/Users/chenlai/pytorch/build_android/aten/src;/Users/chenlai/pytorch/torch/csrc;/Users/chenlai/pytorch/third_party/miniz-2.0.8;/Users/chenlai/pytorch/build_android/caffe2/aten/src/TH;/Users/chenlai/pytorch/aten/src/TH;/Users/chenlai/pytorch/aten/src;/Users/chenlai/pytorch/build_android/caffe2/aten/src;/Users/chenlai/pytorch/build_android/aten/src;/Users/chenlai/pytorch/aten/src;/Users/chenlai/pytorch/aten/../third_party/catch/single_include;/Users/chenlai/pytorch/aten/src/ATen/..;/Users/chenlai/pytorch/build_android/caffe2/aten/src/ATen;/Users/chenlai/pytorch/third_party/miniz-2.0.8;/Users/chenlai/pytorch/caffe2/
torch_cpu INSTALL_RPATH = $ORIGIN
torch_cpu INSTALL_RPATH_USE_LINK_PATH = TRUE
torch_cpu INTERFACE_INCLUDE_DIRECTORIES = $<INSTALL_INTERFACE:include>
torch_cpu INTERFACE_LINK_LIBRARIES = $<LINK_ONLY:fxdiv>;c10;Threads::Threads;$<LINK_ONLY:eigen_blas>;$<LINK_ONLY:pthreadpool>;$<LINK_ONLY:cpuinfo>;$<LINK_ONLY:pytorch_qnnpack>;$<LINK_ONLY:nnpack>;$<LINK_ONLY:XNNPACK>;$<LINK_ONLY:fp16>;$<LINK_ONLY:log>;$<LINK_ONLY:fmt::fmt-header-only>;$<LINK_ONLY:dl>;$<LINK_ONLY:m>;$<LINK_ONLY:nnpack>;$<LINK_ONLY:cpuinfo>;$<LINK_ONLY:ATEN_CPU_FILES_GEN_LIB>
torch_cpu LIBRARY_OUTPUT_DIRECTORY = /Users/chenlai/pytorch/build_android/lib
torch_cpu LINK_LIBRARIES = fxdiv;c10;Threads::Threads;eigen_blas;pthreadpool;cpuinfo;pytorch_qnnpack;nnpack;XNNPACK;fp16;log;fmt::fmt-header-only;dl;m;nnpack;cpuinfo;ATEN_CPU_FILES_GEN_LIB
torch_cpu NAME = torch_cpu
torch_cpu PCH_WARN_INVALID = ON
torch_cpu POSITION_INDEPENDENT_CODE = TRUE
torch_cpu RUNTIME_OUTPUT_DIRECTORY = /Users/chenlai/pytorch/build_android/bin
torch_cpu SKIP_BUILD_RPATH = FALSE
torch_cpu SOURCES = /Users/chenlai/pytorch/aten/src/ATen/BatchedFallback.cpp;/Users/chenlai/pytorch/aten/src/ATen/BatchedTensorImpl.cpp;/Users/chenlai/pytorch/aten/src/ATen/BatchingRegistrations.cpp;/Users/chenlai/pytorch/aten/src/ATen/CPUGeneratorImpl.cpp;/Users/chenlai/pytorch/aten/src/ATen/Context.cpp;/Users/chenlai/pytorch/aten/src/ATen/DLConvertor.cpp;/Users/chenlai/pytorch/aten/src/ATen/DynamicLibrary.cpp;/Users/chenlai/pytorch/aten/src/ATen/ExpandUtils.cpp;/Users/chenlai/pytorch/aten/src/ATen/LegacyTHFunctionsCPU.cpp;/Users/chenlai/pytorch/aten/src/ATen/MemoryOverlap.cpp;/Users/chenlai/pytorch/aten/src/ATen/NamedTensorUtils.cpp;/Users/chenlai/pytorch/aten/src/ATen/ParallelCommon.cpp;/Users/chenlai/pytorch/aten/src/ATen/ParallelNative.cpp;/Users/chenlai/pytorch/aten/src/ATen/ParallelNativeTBB.cpp;/Users/chenlai/pytorch/aten/src/ATen/ParallelOpenMP.cpp;/Users/chenlai/pytorch/aten/src/ATen/ParallelThreadPoolNative.cpp;/Users/chenlai/pytorch/aten/src/ATen/ScalarOps.cpp;/Users/chenlai/pytorch/aten/src/ATen/SequenceNumber.cpp;/Users/chenlai/pytorch/aten/src/ATen/SparseTensorImpl.cpp;/Users/chenlai/pytorch/aten/src/ATen/SparseTensorUtils.cpp;/Users/chenlai/pytorch/aten/src/ATen/TensorGeometry.cpp;/Users/chenlai/pytorch/aten/src/ATen/TensorIndexing.cpp;/Users/chenlai/pytorch/aten/src/ATen/TensorIterator.cpp;/Users/chenlai/pytorch/aten/src/ATen/TensorMeta.cpp;/Users/chenlai/pytorch/aten/src/ATen/TensorNames.cpp;/Users/chenlai/pytorch/aten/src/ATen/TensorUtils.cpp;/Users/chenlai/pytorch/aten/src/ATen/ThreadLocalState.cpp;/Users/chenlai/pytorch/aten/src/ATen/Utils.cpp;/Users/chenlai/pytorch/aten/src/ATen/Version.cpp;/Users/chenlai/pytorch/aten/src/ATen/VmapMode.cpp;/Users/chenlai/pytorch/aten/src/ATen/VmapModeRegistrations.cpp;/Use
```

Reply:
It's probably easier to move these debug changes to a separate pr. I create https://github.com/pytorch/pytorch/pull/52093 instead. Hope it will be easier to review.

Reply:
https://github.com/pytorch/pytorch/pull/52093 is created and closed for this specific change.